### PR TITLE
Improve documentation to taskRunEval

### PR DESCRIPTION
**Pull request by TimSangster**
Changes no functionality, only improves the documentation of taskRunEval

### CODE REVIEW

File: pythonwhat/tasks.py [hunk 1]
```diff
@@ -362,41 +362,73 @@ def taskRunEval(
     tempname="_evaluation_object_",
     call=None,
 ):
+    """
+    Eval an expression tree or node (with setting envs, pre_code and/or expr_code)
+    Utility function later wrapped to extract either result (end state of a variable), output (stdout) or error
+
+    Args:
+        tree (ast): current focused ast, used to get code to execute
+        process: manages shell (see locals.py)
```
Comment (hermansje) [inline pythonwhat/tasks.py]: local.py

File: pythonwhat/tasks.py [hunk 2]
```diff
@@ -362,41 +362,73 @@ def taskRunEval(
...
+        shell: link to to get process namespace from execution up until now
+        env: variables to be replaced in focused code by name
```
Comment (hermansje) [inline pythonwhat/tasks.py]: update value is more clear (replace could be text)

File: pythonwhat/tasks.py [hunk 3]
```diff
@@ -362,41 +362,73 @@ def taskRunEval(
     tempname="_evaluation_object_",
     call=None,
 ):
+    """
+    Eval an expression tree or node (with setting envs, pre_code and/or expr_code)
```
Comment (hermansje) [inline pythonwhat/tasks.py]: Could 'or node' be dropped?

File: pythonwhat/tasks.py [hunk 4]
```diff
@@ -362,41 +362,73 @@ def taskRunEval(
...
+        # Unpack 'container nodes' before checking if a deepcopy is needed
         if isinstance(tree, ast.Module):
             tree = tree.body
         if isinstance(tree, ast.Expression):
             tree = tree.body
         if isinstance(tree, ast.Expr):
             tree = tree.value
+
+        # Avoid a deep copy if specified or if the ast node type indicates we are looking up a variable by name
+        # ast.Name, ast.Subscript and ast.Load do not have side effects in the environment so a deepcopy is not needed
```
Comment (hermansje) [inline pythonwhat/tasks.py]: (most of the time, hence the possibility to override: custom getters could have side effects, e.g a usage counter)

File: pythonwhat/tasks.py (hunk 1 above)
Comment (TimSangster) [reply pythonwhat/tasks.py]: Fixed

File: pythonwhat/tasks.py (hunk 2 above)
Comment (TimSangster) [reply pythonwhat/tasks.py]: Changed to "update value in focused code by name", correct?

File: pythonwhat/tasks.py (hunk 3 above)
Comment (TimSangster) [reply pythonwhat/tasks.py]: Fixed

File: pythonwhat/tasks.py (hunk 4 above)
Comment (TimSangster) [reply pythonwhat/tasks.py]: Fixed

**Pull request merged**